updated_tictactoe.py

```diff
@@ -0,0 +1,146 @@
+import sys
+
+
+def format_user(element):
+    """
+    Return a string representation of a board position that can be logged to console.
+    """
+    if element is None:
+        return '_'
+    elif element:
+        return 'x'
+    else:
+        return 'o'
+
+
+def print_board(board):
+    """
+    Print the board to console.
+    """
+    for row in board:
+        for position in row:
+            print(f"{format_user(position)} ", end="")
+        print()
+
+
+def get_user_move():
+    """
+    Get the users move as an input from the console.  This method valids that the number
+    is between 1 - 9 or "q" (for quiting the game).
+    """
+    user_input = input("Please enter a position 1 through 9 or enter \"q\" to quit: ")
+
+    # Simply quit the program.
+    if user_input == 'q':
+        sys.exit()
+
+    # Validate the number is in range 1 - 9.
+    if user_input in [str(i) for i in range(1, 10)]:
+        return int(user_input) - 1
+    else:
+        # Recurrive call this function to get the user to make a valid move.
+        return get_user_move()
+
+
+def get_board_coords(move):
+    """
+    Return the users move (int 1 - 9) into a tuple defining the index of the row and
+    column in the board.
+    """
+    # Get the row of the move position using floor division (rounds number down to int).
```

Comment:
How does floor division work? So I understand that it divides then rounds down to the nearest integer. So this works for user inputs 1 and 2 which round down to 0 but not for 3 which should be 1. Because it results in 1 it would be a different row.

Reply:
I may have introduced a bug here - my logic may not be correct. Your reasoning is sound